Userland/Libraries/LibGUI/TextEditor.cpp

```diff
@@ -761,13 +761,18 @@ void TextEditor::keydown_event(KeyEvent& event)
                 on_shift_return_pressed();
             return;
         }
-
+        if (event.modifiers() == KeyModifier::Mod_Ctrl && event.key() == KeyCode::Key_Return) {
+            if (on_control_return_pressed)
+                on_control_return_pressed();    
+            return;
+        }
```

Comment:
I'm not sure if it's better to keep this mirrored with `on_shift_return_pressed` above, but this might be less surprising to use if it falls back to the "normal" handler.
```suggestion
        if (event.modifiers() == KeyModifier::Mod_Ctrl && event.key() == KeyCode::Key_Return) {
            if (on_control_return_pressed) {
                on_control_return_pressed();    
                return;
            }
            // If on_control_return_pressed is not used, we fall back to the `on_return_pressed` callback
        }
```